### PR TITLE
[feat]: load json file from storage on startup

### LINKED ISSUE #53: Implement wireguard storage as a json file
Currently all wireguard public keys and what they belong to is held completely in memory. We need to start storing the data so that we can persist everything across process restarts.

We can do this at the same time we make an edit to the map and then persist it to a file.

### PR DESCRIPTION
**Pull request by ksankeerth**
Node manager and cluster manager load data from the from the file system. 
Currently, We use current working directory to store JSON

Fixes #53

@clly , Please check this PR and let me know if anything to be changed. Thanks

### CODE REVIEW

**Comment by clly** [comment]
Hi @ksankeerth. Thank you so much for your contribution!

It looks like it's incomplete though. With this change, both the cluster-manager and node-manager to start persisting the in memory wireguard key stores to disk. However on start up we don't load what was once on disk into memory.

I would look at detecting the presence of the file on mapDB creation. If it exists marshal it into the mapDB.

Let me know if you have any questions!

**Comment by ksankeerth** [comment]
@clly Thanks for suggestion. I have fixed the code to read data from file if already exists.

https://github.com/clly/wireguard-cni/blob/6e62403fdea542ddff95ad496c1ff17665f29a3f/pkg/server/server.go#L114-L126

File: pkg/server/server.go [hunk 1]
```diff
@@ -110,7 +111,17 @@ const clusterWireguardFile = "cluster-wireguard.json"
...
+		if err == nil {
+			err := json.Unmarshal(bytes, &(db).db)
+			if err != nil {
+				return err
+			}
+		} else if errors.Is(err, os.ErrNotExist) {
+			if err := os.MkdirAll(dataDir, 0755); err != nil {
+				return err
+			}
+		} else {
```
Comment (clly) [inline pkg/server/server.go:124]: [nit]: We could do an early return to break up how many if branches we have and reduce the cognitive load.

**Comment by clly** [comment]
@ksankeerth Thank you so much! The change looks good.

The only thing we're missing is a unit test validating these changes.

**Comment by ksankeerth** [comment]
> @ksankeerth Thank you so much! The change looks good.
> 
> The only thing we're missing is a unit test validating these changes.

Thanks for reviewing. Sure, I'll push another commit with tests.

**Comment by ksankeerth** [comment]
@clly I have updated the branch with tests. Please check.Thanks

**Comment by clly** [comment]
Great job! Thank you very much.

**Pull request merged**